### PR TITLE
Support Min/Max for Timestamp

### PR DESCRIPTION
**Pull request by sirpkt**
Related with #3298.
- Add an extension which provides two aggregator `timeMax` and `timeMin`.
- Internally, it stores timestamp value as Long so that we can have DateTime with group by query by finalizeComputing() but only get Long value with select query.
- To support parsing of string represented timestamp, TimestampSpec is slightly modified.
- Incremental index is also modified to accept many different type of timestamp at ingestion time.

### CODE REVIEW

**Comment by fjy** [comment]
@sirpkt does this conflict with https://github.com/druid-io/druid/pull/3168?

Can we get some docs on what timeMax and timeMin do? I'm not entirely clear from reading the description

File: extensions-contrib/time-min-max/src/test/resources/druid.sample.tsv [hunk 1]
```diff
@@ -0,0 +1,1209 @@
+2011-01-12T01:00:00.000Z	spot	automotive	preferred	apreferred	100.000000
```
Comment (fjy) [inline extensions-contrib/time-min-max/src/test/resources/druid.sample.tsv]: can we gzip this file?

**Comment by sirpkt** [comment]
@fjy This is different from #3168.
Its resolution is the same as query granularity because  `__time` is, while this patch returns exact min/max value.
#3168 returns one min/max time for the entire group of (maybe filtered) data while this patch returns separate min/max time for each row of aggregated data.

**Comment by fjy** [comment]
👍

**Comment by jon-wei** [comment]
I think this still needs docs, looks good to me aside from that

**Comment by fjy** [comment]
@sirpkt can we add docs and also an entry in the list of extensions?

**Comment by sirpkt** [comment]
@fjy I added docs.

**Pull request merged**